amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/StreamIdentifier.java

```diff
@@ -16,38 +16,37 @@
 package software.amazon.kinesis.common;
 
 import com.google.common.base.Joiner;
+import lombok.AccessLevel;
+import lombok.Builder;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
-import lombok.NonNull;
 import lombok.experimental.Accessors;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.utils.Validate;
 
 import java.util.Optional;
 import java.util.regex.Pattern;
 
-@EqualsAndHashCode @Getter @Accessors(fluent = true)
+@Builder(access = AccessLevel.PRIVATE)
+@EqualsAndHashCode
+@Getter
+@Accessors(fluent = true)
 public class StreamIdentifier {
-    private final Optional<String> accountIdOptional;
+    @Builder.Default
+    private final Optional<String> accountIdOptional = Optional.empty();
     private final String streamName;
-    private final Optional<Long> streamCreationEpochOptional;
+    @Builder.Default
+    private final Optional<Long> streamCreationEpochOptional = Optional.empty();
+    @Builder.Default
+    private Optional<Arn> streamARNOptional = Optional.empty();
 
     private static final String DELIMITER = ":";
-    private static final Pattern PATTERN = Pattern.compile(".*" + ":" + ".*" + ":" + "[0-9]*");
-
-    private StreamIdentifier(@NonNull String accountId, @NonNull String streamName, @NonNull Long streamCreationEpoch) {
-        this.accountIdOptional = Optional.of(accountId);
-        this.streamName = streamName;
-        this.streamCreationEpochOptional = Optional.of(streamCreationEpoch);
-    }
-
-    private StreamIdentifier(@NonNull String streamName) {
-        this.accountIdOptional = Optional.empty();
-        this.streamName = streamName;
-        this.streamCreationEpochOptional = Optional.empty();
-    }
+    private static final Pattern PATTERN = Pattern.compile(".*" + ":" + ".*" + ":" + "[0-9]*" + "([a-z]{2}(-gov)?-[a-z]+-\\d{1})?");
```

Comment:
> Also, this pattern is most likely incorrect. All * (i.e., zero-or-more) should likely be + (i.e., one-or-more). Not your responsibility to change this, but mentioning it nonetheless.

Agreed, I am quite hesitant to modify the existing patterns

> Should there be a : before the new region token? (This might be a reason to do the String.join.)

Not really, the `*` preceding the region matches the `:` used as a delimiter for the region.